src/UI/molecules/Button.tsx

```diff
@@ -0,0 +1,193 @@
+    >
+      <Corners isActive={isActive} />
+      <Button.InnerContainer ref={buttonInnerContainerRef}>
+        <SpacingContainer paddingRight={buttonSpacing} paddingLeft={buttonSpacing}>
+          <FlexContainer
+            flexFlow="row wrap"
+          > 
+            <ButtonText buttonText={buttonText} size={size} />
+            <ButtonIcon iconName={iconName} size={size} />
+          </FlexContainer>
+        </SpacingContainer>
+      </Button.InnerContainer>
+    </Button.Container>
+  );
+
+  function mapSizeToButtonContainerProps(): ButtonContainerProps {
+    switch (size) {
+    case "small":
+      return {
+        height: "spacing40",
+        width: "auto"
+      };
+    case "medium":
+      return {
+        height: "spacing48",
+        width: "auto"
+      };
+    case "large":
+      return {
+        height: "spacing56",
+        width: "100%"
+      };
+    }
+  }
+
+  function mapTypeToButtonContainerProps(): ButtonContainerProps {
+    switch (type) {
+    case "primary":
+      return {
+        backgroundColor: "blue500",
+        border: "none"
+      };
+    case "secondary":
+      return {
+        backgroundColor: "transparent",
+        border: "1px solid",
+        borderColor: "blue500"
+      };
+    }
+  }
+
+  function handleMouseEnter(): void {
+    setIsActive(true);
+  }
+
+  function handleMouseLeave(): void {
+    setIsActive(false);
+  }
+
+  function handleButtonClick(event: React.MouseEvent<HTMLButtonElement>): void {
+    event.preventDefault();
+
+    if (buttonInnerContainerRef.current) {
+      const { clientX, clientY }: React.MouseEvent = event;
+      const { x, y } = buttonInnerContainerRef.current.getBoundingClientRect();
+
+      const span = document.createElement("span");
+      span.classList.add("ripple");
+      span.style.left = `${clientX - x}px`;
+      span.style.top = `${clientY - y}px`;
+
+      buttonInnerContainerRef.current.appendChild(span);
+
+      setTimeout(() => {
+        span.remove();
+      }, 1000);
+    }
+  }
+}
+
+Button.InnerContainer = styled.div`
+  align-items: center;
+  display: flex;
+  height: 100%;
+  overflow: hidden;
+  position: relative;
+`;
+
+const ripple = keyframes`
+  from {
+    opacity: 1;
+    transform: scale(0);
+  }
+  to {
+    opacity: 0;
+    transform: scale(10);
+  }
+`;
+
+Button.Container = styled.button<ButtonContainerProps>`
+  ${({
+    height,
+    width,
+    backgroundColor,
+    border,
+    borderColor,
+    isActive,
+    theme: {
+      spacing: spacingVariables,
+      colorPalette: { white, blue200 },
+      transitionTimes: { fast },
+      easing: { easeInOut }
+    }
+  }): FlattenSimpleInterpolation => css`
+    background-color: ${(backgroundColor in colorPalette && colorPalette[backgroundColor]) || backgroundColor};
+    border: ${border};
+    border-color: ${colorPalette[borderColor]};
+    box-shadow: ${isActive && `inset 0px 0px 16px 0px ${transparentize(0.5, blue200)}`};
```

Comment:
```suggestion
    box-shadow: `inset 0px 0px 16px 0px ${transparentize(0.5, blue200)}`;
```
This style should be added via `&:hover, &:active, &:focus` - not via `isActive` prop